test/unit-tests.c

```diff
@@ -560,13 +570,20 @@ test_vfu_ctx_create(void **state __attribute__((unused)))
     vfu_ctx = vfu_create_ctx(VFU_TRANS_SOCK, "", LIBVFIO_USER_FLAG_ATTACH_NB,
                              NULL, VFU_DEV_TYPE_PCI);
     assert_non_null(vfu_ctx);
+
     assert_int_equal(1, vfu_ctx->irq_count[VFU_DEV_ERR_IRQ]);
     assert_int_equal(1, vfu_ctx->irq_count[VFU_DEV_REQ_IRQ]);
     assert_int_equal(0, vfu_pci_init(vfu_ctx, VFU_PCI_TYPE_CONVENTIONAL,
                         PCI_HEADER_TYPE_NORMAL, 0));
     assert_int_equal(PCI_STD_HEADER_SIZEOF,
                      vfu_pci_add_capability(vfu_ctx, 0, 0, &pm));
     assert_int_equal(0, vfu_realize_ctx(vfu_ctx));
+
+    patch(close);
```

Comment:
I do not see close() being called in this UT, apart from dma_controller_destroy()(but we do not have DMA regions)
Who can call close() here?

On other note, I think close(vfu_ctx->fd) is missing in vfu_destroy_ctx()

Reply:
This is the failure:

```
1: [ RUN      ] test_vfu_ctx_create
1: [  FAILED  ] test_vfu_ctx_create
1: [  ERROR   ] --- No entries for symbol __wrap_close.
1: /home/john.levon/src/libvfio-user/test/mocks.c:127: error: Could not get value to check parameter fd of function __wrap_close
1: There were no previously declared parameter values for this test.
```

I presume this is from tran->detach() (close_sock())

We can get rid of the vfu_ctx->fd copy I think